app/services/calculate_bursary.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class CalculateBursary
+  class << self
+    # Returns true/false if there are any bursaries available for a given route.
+    def available_for_route?(route)
+      raise_if_route_not_recognised!(route)
+
+      Bursary.find_by(training_route: route)&.bursary_subjects.present?
+    end
+
+    # Returns the amount in pounds of bursary available for a given route/subject combo.
+    def for_route_and_subject(route, subject)
+      raise_if_route_not_recognised!(route)
+
+      # Bursaries are awarded based on the allocation subject for a given
+      # subject specialism.
+      allocation_subject = SubjectSpecialism.find_by!(name: subject).allocation_subject
```

Comment:
I think we may need to use a `where` here so we can filter on additional properties like scholarship. It's likely some bursaries for the same training route will apply to the same allocation subjects but with different amounts.

For eg, there will be two bursaries for provider led each with the same courses they apply to (maths/physics). So querying provider_led with maths could return two records:

One for £9,000 and another for £24,000. Which one do we apply? Well if there's a scholarship available then we should use £24,000 etc. We can probably change this in another PR but just leaving a note here for us to think about

Reply:
Actually ignore above I just realised from the spreadsheet the scenario above isn't possible as there's only one bursary for one route type